pallets/route-executor/src/lib.rs

```diff
@@ -454,16 +466,21 @@ impl<T: Config> Pallet<T> {
 		asset_in: T::AssetId,
 		user_balance_of_asset_in_before_trade: T::Balance,
 		spent_amount: T::Balance,
+		existential_deposit: T::Balance,
 	) -> Result<(), DispatchError> {
-		if spent_amount < user_balance_of_asset_in_before_trade {
-			let user_balance_of_asset_in_after_trade =
-				T::Currency::reducible_balance(asset_in, &who, Preservation::Preserve, Fortitude::Polite);
+		let user_balance_of_asset_in_after_trade =
+			T::Currency::reducible_balance(asset_in, &who, Preservation::Expendable, Fortitude::Polite);
 
-			ensure!(
-				user_balance_of_asset_in_before_trade - spent_amount == user_balance_of_asset_in_after_trade,
-				Error::<T>::InvalidRouteExecution
-			);
-		}
+		let expected_user_balance_of_asset_in_after_trade = user_balance_of_asset_in_before_trade
+			.checked_sub(&spent_amount)
+			.ok_or(Error::<T>::InvalidRouteExecution)?;
+
+		//If the user had leftover less than ED then it is wiped out, hence we can't check the balance precisely
+		ensure!(
+			expected_user_balance_of_asset_in_after_trade < existential_deposit
```

Comment:
We can also check if it's 0

Reply:
You would check if the `user_balance_of_asset_in_after_trade` is equal 0?

Reply:
what is the reason for checking it for 0?